abci/handlers/withdraw/withdraw_test.go

```diff
@@ -19,191 +19,142 @@ func wrapWithdrawTransaction(tx *types.WithdrawTransaction) *types.Transaction {
-				// TODO
-			})
-
-			code := response.WithdrawCheckTxInvalidSignature.Code
-			Convey(fmt.Sprintf("CheckTx should return Code %d", code), func() {
-				res := checkWithdraw(state, rawTx)
-
-				So(res.Code, ShouldEqual, code)
-			})
-
-			code = response.WithdrawDeliverTxInvalidSignature.Code
-			Convey(fmt.Sprintf("DeliverTx should return Code %d", code), func() {
-				res := deliverWithdraw(state, rawTx)
-
-				So(res.Code, ShouldEqual, code)
-			})
-		})
-
-		Convey("If it is an invalid signature format", func() {
-			appCtx.Reset()
-
-			rawTx := wrapWithdrawTransaction(&types.WithdrawTransaction{
-				// TODO
-			})
-
-			code := response.WithdrawCheckTxInvalidSignature.Code
-			Convey(fmt.Sprintf("CheckTx should return Code %d", code), func() {
-				res := checkWithdraw(state, rawTx)
-
-				So(res.Code, ShouldEqual, code)
-			})
-
-			code = response.WithdrawDeliverTxInvalidSignature.Code
-			Convey(fmt.Sprintf("DeliverTx should return Code %d", code), func() {
-				res := deliverWithdraw(state, rawTx)
-
-				So(res.Code, ShouldEqual, code)
-			})
-		})
-
-		Convey("If it is a replayed transaction", func() {
-			appCtx.Reset()
-
-			rawTx := wrapWithdrawTransaction(&types.WithdrawTransaction{
-				// TODO
-			})
-
-			code := response.WithdrawCheckTxDuplicated.Code
-			Convey(fmt.Sprintf("CheckTx should return Code %d", code), func() {
-				res := checkWithdraw(state, rawTx)
+func withdrawTx(from, toAddr []byte, value, fee *big.Int, nonce uint64, sig []byte) *types.WithdrawTransaction {
+	return &types.WithdrawTransaction{
+		From: &types.Identifier{
+			Id: &types.Identifier_LikeChainID{
+				LikeChainID: &types.LikeChainID{
+					Content: from,
+				},
+			},
+		},
+		ToAddr: &types.Address{
+			Content: toAddr,
+		},
+		Value: &types.BigInteger{
+			Content: value.Bytes(),
+		},
+		Fee: &types.BigInteger{
+			Content: fee.Bytes(),
+		},
+		Nonce: nonce,
+		Sig: &types.Signature{
+			Version: 1,
+			Content: sig,
+		},
+	}
+}
 
-				So(res.Code, ShouldEqual, code)
-			})
+func withdrawTxWithEthSender(from *types.Address, toAddr []byte, value, fee *big.Int, nonce uint64, sig []byte) *types.WithdrawTransaction {
+	return &types.WithdrawTransaction{
+		From: &types.Identifier{
+			Id: &types.Identifier_Addr{
+				Addr: from,
+			},
+		},
+		ToAddr: &types.Address{
+			Content: toAddr,
+		},
+		Value: &types.BigInteger{
+			Content: value.Bytes(),
+		},
+		Fee: &types.BigInteger{
+			Content: fee.Bytes(),
+		},
+		Nonce: nonce,
+		Sig: &types.Signature{
+			Version: 1,
+			Content: sig,
+		},
+	}
+}
 
-			code = response.WithdrawDeliverTxDuplicated.Code
-			Convey(fmt.Sprintf("DeliverTx should return Code %d", code), func() {
-				res := deliverWithdraw(state, rawTx)
+func TestValidateWithdrawTransactionFormat(t *testing.T) {
+	zero := big.NewInt(0)
+	one := big.NewInt(1)
 
-				So(res.Code, ShouldEqual, code)
-			})
+	Convey("Given a Withdraw transaction with valid format and LikeChainID sender", t, func() {
```

Comment:
I think the test flow should add a top level Convey to declare tx, and change tx for each sub Convey

Reply:
TODO